Index: BOOTLOADER_Chameleon/BOOTLOADER_Chameleon.cproj
===================================================================
--- BOOTLOADER_Chameleon/BOOTLOADER_Chameleon.cproj	(revision 1392)
+++ BOOTLOADER_Chameleon/BOOTLOADER_Chameleon.cproj	(working copy)
@@ -5,7 +5,7 @@
     <ProjectVersion>6.1</ProjectVersion>
     <ToolchainName>com.Atmel.AVRGCC32.C</ToolchainName>
     <ProjectGuid>{908ad029-5d8f-4abd-bd2b-e01d3fa422a2}</ProjectGuid>
-    <avrdevice>AT32UC3C0512C</avrdevice>
+    <avrdevice>AT32UC3C2512C</avrdevice>
     <avrdeviceseries>uc3c</avrdeviceseries>
     <OutputType>Executable</OutputType>
     <Language>C</Language>
@@ -201,13 +201,13 @@
         <avr32gcc.common.outputfiles.hex>True</avr32gcc.common.outputfiles.hex>
         <avr32gcc.common.outputfiles.lss>True</avr32gcc.common.outputfiles.lss>
         <avr32gcc.common.outputfiles.eep>True</avr32gcc.common.outputfiles.eep>
-        <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
-        <avr32gcc.compiler.symbols.DefSymbols>
-          <ListValues>
-            <Value>BOARD=NUTMEG_BOARD</Value>
-            <Value>_NEW_DATAFLASH</Value>
-          </ListValues>
-        </avr32gcc.compiler.symbols.DefSymbols>
+  <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
+  <avr32gcc.compiler.symbols.DefSymbols>
+    <ListValues>
+      <Value>BOARD=OSCAR_ALPHA</Value>
+      <Value>_NEW_DATAFLASH</Value>
+    </ListValues>
+  </avr32gcc.compiler.symbols.DefSymbols>
         <avr32gcc.compiler.directories.IncludePaths>
           <ListValues>
             <Value>../src/asf/common/services/usb/class/dfu_flip/device/bootloader/avr32/intc</Value>
@@ -296,13 +296,13 @@
             <Value>../src/ASF/avr32/utils/debug</Value>
             <Value>../src/ASF/avr32/drivers/wdt</Value>
             <Value>../src/ASF/common/services/delay</Value>
-            <Value>../src/ASF/avr32/drivers/spi</Value>
-          </ListValues>
-        </avr32gcc.assembler.general.IncludePaths>
-        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-DBOARD=NUTMEG_BOARD -mrelax</avr32gcc.preprocessingassembler.general.AssemblerFlags>
-        <avr32gcc.preprocessingassembler.general.IncludePaths>
-          <ListValues>
-            <Value>../src/asf/common/services/usb/class/dfu_flip/device/bootloader/avr32/intc</Value>
+      <Value>../src/ASF/avr32/drivers/spi</Value>
+    </ListValues>
+  </avr32gcc.assembler.general.IncludePaths>
+  <avr32gcc.preprocessingassembler.general.AssemblerFlags>-DBOARD=OSCAR_BOARD -mrelax</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+  <avr32gcc.preprocessingassembler.general.IncludePaths>
+    <ListValues>
+      <Value>../src/asf/common/services/usb/class/dfu_flip/device/bootloader/avr32/intc</Value>
             <Value>../src/asf/common/services/isp/flip/uc3</Value>
             <Value>../src/asf/avr32/drivers/ast</Value>
             <Value>../src/asf/common/services/usb/class/dfu_flip/device/bootloader/avr32</Value>
@@ -348,7 +348,7 @@
   <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
   <avr32gcc.compiler.symbols.DefSymbols>
     <ListValues>
-      <Value>BOARD=NUTMEG_BOARD</Value>
+      <Value>BOARD=OSCAR_ALPHA</Value>
       <Value>_NEW_DATAFLASH</Value>
     </ListValues>
   </avr32gcc.compiler.symbols.DefSymbols>
@@ -444,13 +444,13 @@
       <Value>../src/ASF/avr32/drivers/wdt</Value>
       <Value>../src/ASF/common/services/delay</Value>
       <Value>../src/ASF/avr32/drivers/spi</Value>
-    </ListValues>
-  </avr32gcc.assembler.general.IncludePaths>
-  <avr32gcc.assembler.debugging.DebugLevel>Default (-g)</avr32gcc.assembler.debugging.DebugLevel>
-  <avr32gcc.preprocessingassembler.general.AssemblerFlags>-DBOARD=NUTMEG_BOARD -DUDD_ENABLE -mrelax</avr32gcc.preprocessingassembler.general.AssemblerFlags>
-  <avr32gcc.preprocessingassembler.general.IncludePaths>
-    <ListValues>
-      <Value>../src/asf/common/services/usb/class/dfu_flip/device/bootloader/avr32/intc</Value>
+          </ListValues>
+        </avr32gcc.assembler.general.IncludePaths>
+        <avr32gcc.assembler.debugging.DebugLevel>Default (-g)</avr32gcc.assembler.debugging.DebugLevel>
+        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-DBOARD=OSCAR_ALPHA -DUDD_ENABLE -mrelax</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+        <avr32gcc.preprocessingassembler.general.IncludePaths>
+          <ListValues>
+            <Value>../src/asf/common/services/usb/class/dfu_flip/device/bootloader/avr32/intc</Value>
       <Value>../src/asf/common/services/isp/flip/uc3</Value>
       <Value>../src/asf/avr32/drivers/ast</Value>
       <Value>../src/asf/common/services/usb/class/dfu_flip/device/bootloader/avr32</Value>
Index: UnitTest_CheckDownloadedImage/UnitTest_CheckDownloadedImage.cproj
===================================================================
--- UnitTest_CheckDownloadedImage/UnitTest_CheckDownloadedImage.cproj	(revision 1392)
+++ UnitTest_CheckDownloadedImage/UnitTest_CheckDownloadedImage.cproj	(working copy)
@@ -67,7 +67,7 @@
         <board id="board.uc3c_ek" value="Add" config="" content-id="Atmel.ASF" />
       </framework-data>
     </AsfFrameworkConfig>
-    <avrdevice>AT32UC3C0512C</avrdevice>
+    <avrdevice>AT32UC3C2512C</avrdevice>
     <avrdeviceseries>uc3c</avrdeviceseries>
     <Language>C</Language>
     <ToolchainName>com.Atmel.AVRGCC32.C</ToolchainName>
@@ -128,9 +128,10 @@
         <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
         <avr32gcc.compiler.symbols.DefSymbols>
           <ListValues>
-            <Value>BOARD=NUTMEG_BOARD</Value>
+            <Value>BOARD=OSCAR_ALPHA</Value>
             <Value>TEST_SUITE_DEFINE_ASSERT_MACRO</Value>
             <Value>_ASSERT_ENABLE_</Value>
+            <Value>_NEW_DATAFLASH</Value>
           </ListValues>
         </avr32gcc.compiler.symbols.DefSymbols>
         <avr32gcc.compiler.directories.IncludePaths>
@@ -179,7 +180,7 @@
             <Value>../src/ASF/common/utils</Value>
           </ListValues>
         </avr32gcc.assembler.general.IncludePaths>
-        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=NUTMEG_BOARD -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=OSCAR_ALPHA -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
         <avr32gcc.preprocessingassembler.general.IncludePaths>
           <ListValues>
             <Value>../src</Value>
@@ -212,7 +213,7 @@
   <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
   <avr32gcc.compiler.symbols.DefSymbols>
     <ListValues>
-      <Value>BOARD=NUTMEG_BOARD</Value>
+      <Value>BOARD=OSCAR_ALPHA</Value>
       <Value>TEST_SUITE_DEFINE_ASSERT_MACRO</Value>
       <Value>_ASSERT_ENABLE_</Value>
       <Value>_NEW_DATAFLASH</Value>
@@ -296,7 +297,7 @@
       <Value>../src/ASF/common/utils</Value>
     </ListValues>
   </avr32gcc.assembler.general.IncludePaths>
-  <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=NUTMEG_BOARD -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+  <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=OSCAR_ALPHA -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
   <avr32gcc.preprocessingassembler.general.IncludePaths>
     <ListValues>
       <Value>../src</Value>
Index: UnitTest_DecryptProgram/UnitTest_DecryptProgram.cproj
===================================================================
--- UnitTest_DecryptProgram/UnitTest_DecryptProgram.cproj	(revision 1392)
+++ UnitTest_DecryptProgram/UnitTest_DecryptProgram.cproj	(working copy)
@@ -128,9 +128,10 @@
         <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
         <avr32gcc.compiler.symbols.DefSymbols>
           <ListValues>
-            <Value>BOARD=NUTMEG_BOARD</Value>
+            <Value>BOARD=OSCAR_ALPHA</Value>
             <Value>TEST_SUITE_DEFINE_ASSERT_MACRO</Value>
             <Value>_ASSERT_ENABLE_</Value>
+            <Value>_NEW_DATAFLASH</Value>
           </ListValues>
         </avr32gcc.compiler.symbols.DefSymbols>
         <avr32gcc.compiler.directories.IncludePaths>
@@ -179,7 +180,7 @@
             <Value>../src/ASF/common/utils</Value>
           </ListValues>
         </avr32gcc.assembler.general.IncludePaths>
-        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=NUTMEG_BOARD -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=OSCAR_ALPHA -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
         <avr32gcc.preprocessingassembler.general.IncludePaths>
           <ListValues>
             <Value>../src</Value>
@@ -212,7 +213,7 @@
   <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
   <avr32gcc.compiler.symbols.DefSymbols>
     <ListValues>
-      <Value>BOARD=NUTMEG_BOARD</Value>
+      <Value>BOARD=OSCAR_ALPHA</Value>
       <Value>TEST_SUITE_DEFINE_ASSERT_MACRO</Value>
       <Value>_ASSERT_ENABLE_</Value>
       <Value>_NEW_DATAFLASH</Value>
@@ -301,7 +302,7 @@
       <Value>../../BOOTLOADER_Chameleon/src/encrypt_decrypt</Value>
     </ListValues>
   </avr32gcc.assembler.general.IncludePaths>
-  <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=NUTMEG_BOARD -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+  <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=OSCAR_ALPHA -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
   <avr32gcc.preprocessingassembler.general.IncludePaths>
     <ListValues>
       <Value>../src</Value>
Index: UnitTest_Image_to_DataFlash/UnitTest_Image_to_DataFlash.cproj
===================================================================
--- UnitTest_Image_to_DataFlash/UnitTest_Image_to_DataFlash.cproj	(revision 1392)
+++ UnitTest_Image_to_DataFlash/UnitTest_Image_to_DataFlash.cproj	(working copy)
@@ -67,7 +67,7 @@
         <board id="board.uc3c_ek" value="Add" config="" content-id="Atmel.ASF" />
       </framework-data>
     </AsfFrameworkConfig>
-    <avrdevice>AT32UC3C0512C</avrdevice>
+    <avrdevice>AT32UC3C2512C</avrdevice>
     <avrdeviceseries>uc3c</avrdeviceseries>
     <Language>C</Language>
     <ToolchainName>com.Atmel.AVRGCC32.C</ToolchainName>
@@ -128,9 +128,10 @@
         <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
         <avr32gcc.compiler.symbols.DefSymbols>
           <ListValues>
-            <Value>BOARD=NUTMEG_BOARD</Value>
+            <Value>BOARD=OSCAR_ALPHA</Value>
             <Value>TEST_SUITE_DEFINE_ASSERT_MACRO</Value>
             <Value>_ASSERT_ENABLE_</Value>
+            <Value>_NEW_DATAFLASH</Value>
           </ListValues>
         </avr32gcc.compiler.symbols.DefSymbols>
         <avr32gcc.compiler.directories.IncludePaths>
@@ -179,7 +180,7 @@
             <Value>../src/ASF/common/utils</Value>
           </ListValues>
         </avr32gcc.assembler.general.IncludePaths>
-        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=NUTMEG_BOARD -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=OSCAR_ALPHA -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
         <avr32gcc.preprocessingassembler.general.IncludePaths>
           <ListValues>
             <Value>../src</Value>
@@ -212,7 +213,7 @@
   <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
   <avr32gcc.compiler.symbols.DefSymbols>
     <ListValues>
-      <Value>BOARD=NUTMEG_BOARD</Value>
+      <Value>BOARD=OSCAR_ALPHA</Value>
       <Value>TEST_SUITE_DEFINE_ASSERT_MACRO</Value>
       <Value>_ASSERT_ENABLE_</Value>
       <Value>_NEW_DATAFLASH</Value>
@@ -296,7 +297,7 @@
       <Value>../src/ASF/common/utils</Value>
     </ListValues>
   </avr32gcc.assembler.general.IncludePaths>
-  <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=NUTMEG_BOARD -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+  <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=OSCAR_ALPHA -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
   <avr32gcc.preprocessingassembler.general.IncludePaths>
     <ListValues>
       <Value>../src</Value>
Index: UnitTest_Image_to_DataFlash_ErrorState/UnitTest_Image_to_DataFlash_ErrorState.cproj
===================================================================
--- UnitTest_Image_to_DataFlash_ErrorState/UnitTest_Image_to_DataFlash_ErrorState.cproj	(revision 1392)
+++ UnitTest_Image_to_DataFlash_ErrorState/UnitTest_Image_to_DataFlash_ErrorState.cproj	(working copy)
@@ -67,7 +67,7 @@
         <board id="board.uc3c_ek" value="Add" config="" content-id="Atmel.ASF" />
       </framework-data>
     </AsfFrameworkConfig>
-    <avrdevice>AT32UC3C0512C</avrdevice>
+    <avrdevice>AT32UC3C2512C</avrdevice>
     <avrdeviceseries>uc3c</avrdeviceseries>
     <Language>C</Language>
     <ToolchainName>com.Atmel.AVRGCC32.C</ToolchainName>
@@ -127,9 +127,10 @@
         <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
         <avr32gcc.compiler.symbols.DefSymbols>
           <ListValues>
-            <Value>BOARD=NUTMEG_BOARD</Value>
+            <Value>BOARD=OSCAR_ALPHA</Value>
             <Value>TEST_SUITE_DEFINE_ASSERT_MACRO</Value>
             <Value>_ASSERT_ENABLE_</Value>
+			<Value>_NEW_DATAFLASH</Value>
           </ListValues>
         </avr32gcc.compiler.symbols.DefSymbols>
         <avr32gcc.compiler.directories.IncludePaths>
@@ -178,7 +179,7 @@
             <Value>../src/ASF/common/utils</Value>
           </ListValues>
         </avr32gcc.assembler.general.IncludePaths>
-        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=NUTMEG_BOARD -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=OSCAR_ALPHA -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
         <avr32gcc.preprocessingassembler.general.IncludePaths>
           <ListValues>
             <Value>../src</Value>
@@ -211,7 +212,7 @@
         <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
         <avr32gcc.compiler.symbols.DefSymbols>
           <ListValues>
-            <Value>BOARD=NUTMEG_BOARD</Value>
+            <Value>BOARD=OSCAR_ALPHA</Value>
             <Value>TEST_SUITE_DEFINE_ASSERT_MACRO</Value>
             <Value>_ASSERT_ENABLE_</Value>
             <Value>_NEW_DATAFLASH</Value>
@@ -295,7 +296,7 @@
             <Value>../src/ASF/common/utils</Value>
           </ListValues>
         </avr32gcc.assembler.general.IncludePaths>
-        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=NUTMEG_BOARD -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=OSCAR_ALPHA -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
         <avr32gcc.preprocessingassembler.general.IncludePaths>
           <ListValues>
             <Value>../src</Value>
Index: UnitTest_Image_to_DataFlash_InvalidImage/UnitTest_Image_to_DataFlash_InvalidImage.cproj
===================================================================
--- UnitTest_Image_to_DataFlash_InvalidImage/UnitTest_Image_to_DataFlash_InvalidImage.cproj	(revision 1392)
+++ UnitTest_Image_to_DataFlash_InvalidImage/UnitTest_Image_to_DataFlash_InvalidImage.cproj	(working copy)
@@ -67,7 +67,7 @@
         <board id="board.uc3c_ek" value="Add" config="" content-id="Atmel.ASF" />
       </framework-data>
     </AsfFrameworkConfig>
-    <avrdevice>AT32UC3C0512C</avrdevice>
+    <avrdevice>AT32UC3C2512C</avrdevice>
     <avrdeviceseries>uc3c</avrdeviceseries>
     <Language>C</Language>
     <ToolchainName>com.Atmel.AVRGCC32.C</ToolchainName>
@@ -128,9 +128,10 @@
         <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
         <avr32gcc.compiler.symbols.DefSymbols>
           <ListValues>
-            <Value>BOARD=NUTMEG_BOARD</Value>
+            <Value>BOARD=OSCAR_ALPHA</Value>
             <Value>TEST_SUITE_DEFINE_ASSERT_MACRO</Value>
             <Value>_ASSERT_ENABLE_</Value>
+			<Value>_NEW_DATAFLASH</Value>
           </ListValues>
         </avr32gcc.compiler.symbols.DefSymbols>
         <avr32gcc.compiler.directories.IncludePaths>
@@ -179,7 +180,7 @@
             <Value>../src/ASF/common/utils</Value>
           </ListValues>
         </avr32gcc.assembler.general.IncludePaths>
-        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=NUTMEG_BOARD -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=OSCAR_ALPHA -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
         <avr32gcc.preprocessingassembler.general.IncludePaths>
           <ListValues>
             <Value>../src</Value>
@@ -212,7 +213,7 @@
         <avr32gcc.common.outputfiles.srec>True</avr32gcc.common.outputfiles.srec>
         <avr32gcc.compiler.symbols.DefSymbols>
           <ListValues>
-            <Value>BOARD=NUTMEG_BOARD</Value>
+            <Value>BOARD=OSCAR_ALPHA</Value>
             <Value>TEST_SUITE_DEFINE_ASSERT_MACRO</Value>
             <Value>_ASSERT_ENABLE_</Value>
             <Value>_NEW_DATAFLASH</Value>
@@ -296,7 +297,7 @@
             <Value>../src/ASF/common/utils</Value>
           </ListValues>
         </avr32gcc.assembler.general.IncludePaths>
-        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=NUTMEG_BOARD -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
+        <avr32gcc.preprocessingassembler.general.AssemblerFlags>-mrelax -DBOARD=OSCAR_ALPHA -DTEST_SUITE_DEFINE_ASSERT_MACRO -D_ASSERT_ENABLE_</avr32gcc.preprocessingassembler.general.AssemblerFlags>
         <avr32gcc.preprocessingassembler.general.IncludePaths>
           <ListValues>
             <Value>../src</Value>
